platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/internal/cc/impl/ConfigurationCacheEncryptionIntegrationTest.groovy

```diff
@@ -149,6 +149,21 @@ class ConfigurationCacheEncryptionIntegrationTest extends AbstractConfigurationC
         EncryptionKind.ENV_VAR  | true    | false
     }
 
+    /**
+     * Returns the bytes as encoded by {@link org.gradle.internal.serialize.kryo.StringDeduplicatingKryoBackedEncoder},
+     * which uses the last byte as a marker.
+     *
+     * @see com.esotericsoftware.kryo.io.Output#writeString(java.lang.String)
+     */
+    @SuppressWarnings('GrDeprecatedAPIUsage')
+    private static byte[] encodedBytesOf(String string) {
```

Comment:
🤔 TBH, I think this test now knows too much of the encoding details and has become fragile.

Reply:
Indeed. It can still be useful and less fragile by having the asserts ignoring the last character but I'd deal with it later once we introduce compression.